methods/did:.md

```diff
@@ -0,0 +1,33 @@
+# did:*
+
+The "any" DID method encoding allows any DID to be represented as a multidid by essentially putting the entire DID ASCII string in the URL portion of the multidid byte string.
+
+## Method multicodec
+
+`0x55`
+
+## Format
+
+```
+<multidid-code><method-code><method-bytes><url-length><url-bytes>
+```
+
+Where:
+
+* `multidid-code` - a varint encoding of `0xd1d`
+* `method-code` - a varint encoding of `0x55`
+* `method-bytes` - MUST contain exactly zero bytes
+* `url-length` - a varint describing the length of the `url-bytes` parameter
+* `url-bytes` - a UTF-8 encoded string representing the DID and URL component (e.g. everything except `did:`)
+
+## Examples
+
+Displayed using base 16.
+
+```
+// did:example:123456
+0x9d1a370e6578616d706c653a313233343536
```

Comment:
@oed 
https://github.com/chrisdickinson/varint seems to be consistent with the example strings.

```sh
npm i varint
```
```js
var varint = require('varint')
```

<multidid-code><method-code><method-bytes><url-length><url-bytes>

did:example:123456
0x9d1a 37 0e 6578616d706c653a313233343536

multidid-code 0x0d1d 
```
varint.encode(0x0d1d).map((x) => x.toString(16)).join("")  // '9d1a'
```

method-code Raw 55  // 0x37
```js
varint.encode(55).map((x) => x.toString(16)).join("")  // '37'
```

method-bytes ""  // raw has no method-bytes

url-length  // 0x0e
```js
"example:123456".length  // 14
varint.encode(14).map((x) => x.toString(16)).join("") // "e"
// warn: need to include the '0' padding 0x0e
```

url-bytes 0x6578616d706c653a313233343536
```js
Array.from("example:123456", (x) => x.charCodeAt(0).toString(16)).join("")
// '6578616d706c653a313233343536'
```


// did:example:123456?versionId=1
0x9d1a 37 1a 6578616d706c653a3132333435363f76657273696f6e49643d31
```js
    0x9d1a  // multidid-code 0x0d1d 
    0x37  // method-code 55 Raw
    0x1a  // url-length 26
    0x6578616d706c653a3132333435363f76657273696f6e49643d31  // "example:123456?versionId=1" url-bytes
```

Reply:
NB @chunningham